jetstream/metadata.py

```diff
@@ -0,0 +1,72 @@
+from google.cloud import storage
+from typing import Dict, List
+
+import attr
+import cattr
+import json
+import logging
+
+from jetstream import bq_normalize_name
+from jetstream.config import AnalysisConfiguration
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class MetricsMetadata:
+    friendly_name: str
+    description: str
+    bigger_is_better: bool
+
+
+@attr.s(auto_attribs=True)
+class ExperimentMetadata:
+    metrics: Dict[str, MetricsMetadata]
+    probesets: Dict[str, List[str]]
+
+    @classmethod
+    def from_config(cls, config: AnalysisConfiguration) -> "ExperimentMetadata":
+        all_metrics = [
+            summary.metric for period, summaries in config.metrics.items() for summary in summaries
+        ]
+        all_metrics_distinct = list(
+            set(all_metrics)
+        )  # some metrics are used in multiple analysis periods
+
+        metrics_metadata = {
+            metric.name: MetricsMetadata(
+                friendly_name=metric.friendly_name,
+                description=metric.description,
+                bigger_is_better=metric.bigger_is_better,
+            )
+            for metric in all_metrics_distinct
+        }
+
+        probesets_metadata = {
+            probe_set.slug: list(set([summary.metric.name for summary in probe_set.to_summaries()]))
```

Comment:
The ADR mentions that this list is supposed to be ordered. _The first metric in the list is assumed to be the most interesting metric to display. It will typically be a conversion metric._ Do you have a preference in what order the metrics should appear here? Currently, we have only `<...>_ever_used` and `<...>_sum` (I guess this might get extended in the future.)

Reply:
Yeah, I think ever_used is the more interesting metric.

I think that the ProbeType classes should probably be responsible for determining this (they do emit ordered metrics) but I realized that I don't know what to do when a ProbeSet has more than one probe or more than one type of probe. It makes sense for Experimenter to dictate the order -- I guess we follow Experimenter's lead.

I'll add this to the probeset design considerations.

Reply:
Anyway, this looks like a fine thing to do except that I'm not sure that set() is specified to be order-preserving; I think these values must always be unique (or we'd have a column collision in the output) so we can skip the list(set(...)).

(You can also write set comprehensions, like `{i for i in range(20)}`)

Reply:
`<...>_sum` would appear multiple times since it's used for calculating `BootstrapMean` and `Deciles`. But using `list(dict.fromkeys())` will preserve the alphabetical order here for now. We should probably come up with a better way to sort these metrics in the future.

Reply:
Oh, right; we're processing summaries, not metrics, thanks.